bindings/swift/src/ffi_functions.rs

```diff
@@ -898,6 +898,41 @@ impl ETOPaySdk {
         .map_err(|err| format!("{:#?}", err))
     }
 
+    /// Estimates the amount of gas required to execute the transaction.
+    ///
+    /// # Arguments
+    ///
+    /// * `pin` - The PIN of the user.
+    /// * `address` - The receiver's address.
+    /// * `amount` - The amount to send.
+    /// * `data` - The associated data with the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * Ok - the gas estimation of the transaction
+    /// * Err - if the user or wallet is not initialized, there is an error verifying the PIN, or there is an error estimating the transaction gas.
+    pub async fn estimate_gas(
+        &self,
+        pin: String,
+        address: String,
+        amount: f64,
+        data: Option<Vec<u8>>,
+    ) -> Result<GasCostEstimation, String> {
+        let mut sdk = self.inner.write().await;
+        async move {
+            async move {
+                let amount = CryptoAmount::try_from(amount)?;
+                let pin = EncryptionPin::try_from_string(pin)?;
+                sdk.estimate_gas(&pin, &address, amount, data).await
+            }
+            .await
+            .map(Into::into)
+            .map_err(|err| format!("{:#?}", err))
+        }
+        .await
+        .map_err(|err| format!("{:#?}", err))
```

Comment:
Can we not remove one level of `async move` here?
```suggestion
        async move {
            let amount = CryptoAmount::try_from(amount)?;
            let pin = EncryptionPin::try_from_string(pin)?;
            sdk.estimate_gas(&pin, &address, amount, data).await
        }
        .await
        .map(Into::into)
        .map_err(|err| format!("{:#?}", err))
```

Reply:
good one :D